boards/slstk3401a/board.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2015-2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_slstk3401a
+ * @{
+ *
+ * @file
+ * @brief       Board specific implementations SLSTK3401A board
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "board_common.h"
+
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* initialize the CPU */
+    cpu_init();
+
+    /* perform common board initialization */
+    board_common_init();
+
+    /* initialize the Si7021 sensor */
+#if SI7021_ENABLED
```

Comment:
use:
```c
#ifdef MODULE_SI70XX
```
instead. This flag is set automatically when one adds USEMODULE += si7021 in the application makefile.

Reply:
Fixed.

Reply:
I didn't see this comment. IMHO this is not needed at all, since the inclusion of the driver is not really optional, the device is physically attached to the board and cannot be detached or deactivated, thus it belongs to the board and should be initialised whenever it's used.

The iotlab-m3 board has also some sensors attached, maybe you can take a look there to see how it's implemented, but as far as I can tell, they are only initialised when they're used through auto_init.

Another option would be to add a params file so you can access the sensor through SAUL using the default example.

Reply:
>  since the inclusion of the driver is not really optional, the device is physically attached to the board

yes it is, if the application doesn't load the driver module, there's no need for initializing and setting the pin that control it on the board.

Reply:
I just got the purpose of this, made my comment in my review.